src/config_file.c

```diff
@@ -668,6 +668,151 @@ static GHashTable *parse_artifact_repos(const char *filename, const char *data_d
 	return g_steal_pointer(&repos);
 }
 
+static const gchar *supported_polling_candidate_criteria[] = {
+	"higher-semver",
+	"different-version",
+	NULL
+};
+static const gchar *default_polling_candidate_criteria[] = {
+	"higher-semver",
+	NULL
+};
+static const gchar *supported_polling_install_criteria[] = {
+	"always",
+	"higher-semver",
+	"different-version",
+	NULL
+};
+static const gchar *supported_polling_reboot_criteria[] = {
+	"updated-slots",
+	"updated-artifacts",
+	"failed-update",
+	NULL
+};
+
+static gboolean parse_polling_config(RaucConfig *config, GKeyFile *key_file, GError **error)
+{
+	GError *ierror = NULL;
+
+	if (!g_key_file_has_group(key_file, "polling"))
+		return TRUE;
+
+	if (!ENABLE_STREAMING) {
+		g_set_error(
+				error,
+				R_CONFIG_ERROR,
+				R_CONFIG_ERROR_POLLING,
+				"Polling not supported, recompile with -Dstreaming=true"
+				);
+		return FALSE;
+	}
+
+	config->polling_inhibit_files = key_file_consume_string_list(key_file, "polling", "inhibit-files", NULL, &ierror);
+	if (ierror) {
+		g_propagate_error(error, ierror);
+		return FALSE;
+	}
+
+	config->polling_candidate_criteria = key_file_consume_string_list(key_file, "polling", "candidate-criteria", supported_polling_candidate_criteria, &ierror);
+	if (ierror) {
+		g_propagate_error(error, ierror);
+		return FALSE;
+	} else if (!config->polling_candidate_criteria) {
+		config->polling_candidate_criteria = g_strdupv((GStrv)default_polling_candidate_criteria);
+	}
+
+	config->polling_install_criteria = key_file_consume_string_list(key_file, "polling", "install-criteria", supported_polling_install_criteria, &ierror);
+	if (ierror) {
+		g_propagate_error(error, ierror);
+		return FALSE;
+	}
+
+	config->polling_reboot_criteria = key_file_consume_string_list(key_file, "polling", "reboot-criteria", supported_polling_reboot_criteria, &ierror);
+	if (ierror) {
+		g_propagate_error(error, ierror);
+		return FALSE;
+	}
+
+	config->polling_source = key_file_consume_string(key_file, "polling", "source", NULL);
```

Comment:
If 'polling_source' needs to be a URL anyway, maybe it might make sense to name it 'polling_url'?  This also reduces the potential confusion with the `GSource` `polling_source`.